java/code/src/com/redhat/rhn/frontend/xmlrpc/recurringaction/RecurringActionHandler.java

```diff
@@ -0,0 +1,243 @@
+                case GROUP:
+                    schedules = RecurringActionManager.listGroupRecurringActions(entityId, loggedInUser);
+                    break;
+                case ORG:
+                    schedules = RecurringActionManager.listOrgRecurringActions(entityId, loggedInUser);
+                    break;
+                default:
+                    throw new IllegalStateException("Unsupported type " + entityType);
+            }
+            return schedules;
+        }
+        catch (PermissionException e) {
+            throw new PermissionCheckFailureException(e.getMessage());
+        }
+    }
+
+    /**
+     * Return recurring action with given action id.
+     *
+     * @param loggedInUser The current user
+     * @param actionId id of the action
+     * @return the list of recurring actions
+     *
+     * @xmlrpc.doc Return recurring action with given action id.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #param_desc("int", "action_id", "Id of the action")
+     * @xmlrpc.returntype $RecurringActionSerializer
+     */
+    public RecurringAction lookupById(User loggedInUser, Integer actionId) {
+        try {
+            RecurringAction action = RecurringActionFactory.lookupById(actionId).orElseThrow();
+            if (action.canAccess(loggedInUser)) {
+                return action;
+            }
+        }
+        catch (NoSuchElementException e) {
+            throw new EntityNotExistsFaultException("Action with id: " + actionId + " does not exist");
+        }
+        throw new PermissionCheckFailureException("Action not accessible to user: " + loggedInUser);
+    }
+
+    /**
+     * Create a new recurring action.
+     *
+     * @param loggedInUser The current user
+     * @param actionProps Map containing action properties
+     * @return action id or exception thrown otherwise
+     *
+     * @xmlrpc.doc Create a new recurring action.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param
+     *  #struct("actionProps")
+     *      #prop_desc("string", "entity_type", "The type of the target entity")
+     *      #prop_desc("string", "entity_id", "The id of the target entity")
+     *      #prop_desc("string", "name", "The name of the action")
+     *      #prop_desc("string", "cron_expr", "The execution frequency of the action")
+     *      #prop_desc("boolean", "test", "Whether the action should be executed in test mode")
+     *  #struct_end()
+     * @xmlrpc.returntype int action_id - The action id of the recurring action
+     */
+    public Long create(User loggedInUser, Map<String, Object> actionProps) {
+        RecurringAction action = createAction(actionProps, loggedInUser);
+        return save(loggedInUser, action);
+    }
+
+    /* Helper method */
+    private RecurringAction createAction(Map<String, Object> actionProps, User user) {
+        RecurringAction action;
+        if (actionProps.containsKey("id") || !actionProps.containsKey("entity_type") ||
```

Comment:
There is already a method for validating such props maps: `validateMap`. It's even inherited from the `BaseHandler`.
I think it does precisely this, so you could simply write:
```java
validateMap(Set.of("id", "entity_type", "entity_id", ...), actionProps);
```

Reply:
Right, but this does only check if only valid keys are provided, not if valid keys are actually there. E.g. you could just omit the `entity_id` in your call and it would not complain.

Reply:
Ah, I see. It works the other way around: validating that the given map doesn't contain an "invalid" key.